src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedStaticConstructor.cs

```diff
@@ -13,6 +13,7 @@ internal sealed class SynthesizedStaticConstructor : MethodSymbol
     {
         private readonly NamedTypeSymbol _containingType;
         private ThreeState _lazyShouldEmit = ThreeState.Unknown;
+        private ThreeState _lazyIsNullableEnabled;
```

Comment:
Did you consider adding a `bool? isNullableEnabled` to the constructor here where `null` meant "look at the fields"? It seems like in all the cases we create this we either 

1. Know the fields are completely added at `new SytnhesizedStaticConstructor(...)` time 
2. Know that nullable analysis is not needed because it doesn't contain user visible code (lambda initialization for example). 

Feel like having the call site dictate whether we scan the fields, and when we scan them, could be more reliable than defaulting to scanning them based on member type in `IsNullableEnabledCore`